app/modules/intelligence/tools/change_detection/change_detection_tool.py

```
@@ -193,118 +354,475 @@ def find_entry_points(self, identifiers, project_id):
+                        }
+                        result = ChangeDetectionResponse(
+                            patches=patches_summary, changes=changes_list
+                        )
+                    else:
+                        result = ChangeDetectionResponse(
+                            patches=patches_dict, changes=changes_list
+                        )
+                    logging.info(
+                        f"[CHANGE_DETECTION] Returning result with {len(patches_dict)} patches and {len(changes_list)} changes"
+                    )
+                    return result
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)